libcrux-ml-kem/src/vector/avx2.rs

```diff
@@ -42,6 +42,218 @@ fn vec_from_i16_array(array: &[i16]) -> SIMD256Vector {
     }
 }
 
+#[inline(always)]
```

Comment:
These uglyficiations are really unfortunate. We should have a tracking issue somewhere to track getting this fixed.

Reply:
You mean the pulling out of functions from an impl? Yes, we hope that the opaque extensions PR or a follow up will help with this: https://github.com/hacspec/hax/pull/1134

Reply:
Yes, there are the inherent impls (https://github.com/cryspen/libcrux/pull/711#discussion_r1879677274) and the trait impls. We should have something for both to track status and progress for them.